src/cdi_CPEloss/Analytic_Eloss_Model.hh

```diff
@@ -0,0 +1,84 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   cdi_CPEloss/Analytic_Eloss_Model.hh
+ * \author Kendra P. Long, Ben R. Ryan
+ * \date   Feb 21 2020
+ * \brief  Analytic_Eloss_Model class definition.
+ * \note   Copyright (C) 2016-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+#ifndef rtt_cdi_CPEloss_Analytic_Eloss_Model_hh
+#define rtt_cdi_CPEloss_Analytic_Eloss_Model_hh
+
+#include "cdi/CPCommon.hh"
+#include "units/PhysicalConstexprs.hh"
+#include <cmath>
+#include <vector>
+
+namespace rtt_cdi_cpeloss {
+//============================================================================//
+/*!
+ * \class Analytic_Eloss_Model
+ * \brief Analytic_Eloss_Model base class.
+ *
+ * This is a base class that defines the interface given to
+ * Analytic_Eloss_Model constructors.  The user
+ * can define any derived model class that will work with these analytic opacity
+ * generation classes as long as it implements the functions required, namely
+ *
+ * \arg double calculate_eloss(double T, double rho)
+ * \arg sf_double get_parameters()
+ *
+ * This class is a pure virtual base class.
+ *
+ * The returned eloss coefficient is a power with units of keV shk^-1.
+ */
+//============================================================================//
+
+class Analytic_Eloss_Model {
+public:
+  // Typedefs.
+  typedef std::vector<char> sf_char;
```

Comment:
I'm sure I copy-pasted these typedefs in from some other class, but maybe it's dumb to have them here. I never know if typedef-ing in the base class is clever, or confusing...

Reply:
Good catch -- these aren't used currently so I just took them out